Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -266,19 +287,21 @@ void OpenALStream::SoundLoop()
     u64 num_samples_to_render =
         (audio_dma_period * ais_samples_per_second) / SystemTimers::GetTicksPerSecond();
 
-    unsigned int numSamples = static_cast<unsigned int>(num_samples_to_render);
-    unsigned int minSamples =
-        surround_capable ? 240 : 0;  // DPL2 accepts 240 samples minimum (FWRDURATION)
+    unsigned int min_samples = static_cast<unsigned int>(num_samples_to_render);
+
+	if (samples_per_buffer < min_samples)
+	{
+      ERROR_LOG(AUDIO, "Current latency too low. Consider increasing it.");
+	}
 
-    numSamples = (numSamples > OAL_MAX_SAMPLES) ? OAL_MAX_SAMPLES : numSamples;
-    numSamples = m_mixer->Mix(realtimeBuffer.data(), numSamples, false);
+	unsigned int rendered_samples = m_mixer->Mix(realtime_buffer.data(), samples_per_buffer, false);
 
     // Convert the samples from short to float
-    std::vector<float> dest(OAL_MAX_SAMPLES * STEREO_CHANNELS);
-    for (u32 i = 0; i < numSamples * STEREO_CHANNELS; ++i)
-      dest[i] = static_cast<float>(realtimeBuffer[i]) / (1 << 15);
+    std::vector<float> dest(rendered_samples * STEREO_CHANNELS);
```

Comment:
Since `rendered_samples` will never be larger than `samplers_per_buffer`, these vectors should be allocated ahead of time.